executor/join.go

```diff
@@ -262,10 +261,7 @@ func (e *HashJoinExec) wait4Inner() (finished bool, err error) {
 // fetchInnerRows fetches all rows from inner executor,
 // and append them to e.innerResult.
 func (e *HashJoinExec) fetchInnerRows(ctx context.Context, chkCh chan<- *chunk.Chunk, doneCh chan struct{}) {
-	defer func() {
-		close(chkCh)
-		e.fetchInnerDone.Done()
-	}()
+	defer close(chkCh)
```

Comment:
Could you explain why `WaitGroup` does not work? thanks

Reply:
this exec is too complex, the situation missed by us is that `finished` flag will make `chkCh`'s consumer goroutine exit before producer one, lastday we found producer are blocked on `chkCh <- chk` then no chance to call `wg.Done` 

maybe it's more suitable consume data(also check closeCh) and channel close to handle this...